components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java

```diff
@@ -100,13 +109,26 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
 
         // get the authenticated username from the GSS Token
         String fullyQualifiedName = getAuthenticatedUserFromToken(gssCredential, Base64.decode(gssToken));
+        String realm = IWAAuthenticationUtil.extractRealmFromUserName(fullyQualifiedName);
         String authenticatedUserName = IWAAuthenticationUtil.getDomainAwareUserName(fullyQualifiedName);
+
         if (log.isDebugEnabled()) {
             log.debug("Authenticated Federated User : " + authenticatedUserName);
         }
 
-        context.setSubject(
-                AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserName));
+        boolean isExistInPrimaryUserStore = isExistsInUserStore(authenticatedUserName,
+                                                                context.getTenantDomain(), userStoreDomain);
+        if (!isExistInPrimaryUserStore) {
+            String msg = "User " + authenticatedUserName + "not found in the user store of tenant " + userStoreDomain;
+            throw new AuthenticationFailedException("Authentication Failed, " + msg);
+        }
+        AuthenticatedUser authenticatedUserObj =
+                AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserName);
```

Comment:
Lets consider federated authenticator behaviour if field is null for user stores. Else if there are any entries lets lookup in user stores and return claims